packages/dx-react-scheduler-demos/src/demo-sources/scheduler-featured-redux-integration/material-ui/demo.jsx

```diff
@@ -0,0 +1,327 @@
+import TextField from '@material-ui/core/TextField';
+import ButtonGroup from '@material-ui/core/ButtonGroup';
+import Button from '@material-ui/core/Button';
+import { withStyles } from '@material-ui/core/styles';
+import { teal, orange, red } from '@material-ui/core/colors';
+import { fade } from '@material-ui/core/styles/colorManipulator';
+import classNames from 'classnames';
+import { ViewState } from '@devexpress/dx-react-scheduler';
+import {
+  Scheduler,
+  WeekView,
+  Toolbar,
+  DateNavigator,
+  Appointments,
+  DayView,
+  ViewSwitcher,
+} from '@devexpress/dx-react-scheduler-material-ui';
+
+import { appointments } from '../../../demo-data/appointments';
+
+const styles = ({ spacing, palette }) => ({
+  flexibleSpace: {
+    margin: '0 auto 0 0',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  textField: {
+    width: '100px',
+    marginLeft: spacing(1),
+    marginTop: 0,
+    marginBottom: 0,
+    height: spacing(4.875),
+  },
+  locationSelector: {
+    marginLeft: spacing(1),
+    height: spacing(4.875),
+  },
+  button: {
+    paddingLeft: spacing(1),
+    paddingRight: spacing(1),
+    width: spacing(10),
+  },
+  selectedButton: {
+    paddingLeft: spacing(1),
+    paddingRight: spacing(1),
+    width: spacing(10),
+    background: palette.primary[400],
+    color: palette.primary[50],
+    '&:hover': {
+      backgroundColor: palette.primary[500],
+    },
+    border: `1px solid ${palette.primary[400]}!important`,
+    borderLeft: `1px solid ${palette.primary[50]}!important`,
+    '&:first-child': {
+      borderLeft: `1px solid ${palette.primary[50]}!important`,
+    },
+  },
+  title: {
+    fontWeight: 'bold',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+  },
+  textContainer: {
+    lineHeight: 1,
+    whiteSpace: 'pre-wrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    width: '100%',
+  },
+  time: {
+    display: 'inline-block',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  text: {
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+  },
+  container: {
+    width: '100%',
+  },
+  weekendCell: {
+    backgroundColor: fade(palette.action.disabledBackground, 0.04),
+    '&:hover': {
+      backgroundColor: fade(palette.action.disabledBackground, 0.04),
+    },
+    '&:focus': {
+      backgroundColor: fade(palette.action.disabledBackground, 0.04),
+    },
+  },
+  weekEnd: {
+    backgroundColor: fade(palette.action.disabledBackground, 0.06),
+  },
+  firstRoomAppointment: {
+    backgroundColor: teal[400],
+    '&:hover': {
+      backgroundColor: teal[500],
+    },
+  },
+  secondRoomAppointment: {
+    backgroundColor: orange[400],
+    '&:hover': {
+      backgroundColor: orange[500],
+    },
+  },
+  thirdRoomAppointment: {
+    backgroundColor: red[400],
+    '&:hover': {
+      backgroundColor: red[500],
+    },
+  },
+});
+
+const LOCATIONS = ['Room 1', 'Room 2', 'Room 3'];
```

Comment:
This information is duplicated, I think we should either retrieve it from appointments or export along with appointment from data